### PR TITLE
HDDS-7794. [snapshot] SnapshotDiff should throw better error messages for exception handling

### PR DESCRIPTION
**Pull request by swamirishi**
## What changes were proposed in this pull request?
Add error message in case snapshot info not found.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-7794
## How was this patch tested?
Existing Unit test

### CODE REVIEW

**Review by prashantpogde — APPROVED** [review]
LGTM. thanks @swamirishi for this PR.

**Comment by smengcl** [comment]
@swamirishi Please fix the PR title. It should use `.` instead of `:`

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotUtils.java [hunk 1]
```diff
@@ -70,7 +70,8 @@ public static SnapshotInfo getSnapshotInfo(final OzoneManager ozoneManager,
       throw e;
     }
     if (snapshotInfo == null) {
-      throw new OMException(KEY_NOT_FOUND);
+      throw new OMException("Snapshot '" + snapshotKey + "' is not found.",
```
Comment (hemantk-12) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotUtils.java:73]: nit: Should it be `snapshotKey` or `snapshotName`? I think `snapshotName` should be better.

Comment (smengcl) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotUtils.java:73]: @hemantk-12 In this case it only has `dbKey` (e.g. result of `SnapshotInfo.getTableKey(volumeName, bucketName, snapshotName)`)

Comment (hemantk-12) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotUtils.java:73]: You are right. But I'm just thinking from client perspective `snapshotName` would be better which can be extracted from the `snapshotKey` I guess. At the same time it is not a big deal to return `snapshotKey` in exception message.

**Pull request merged**